File: Golang/Easy/3487_Maximum_Unique_Subarray_Sum_After_Deletion/solution.go
package maximumuniquesubarraysumafterdeletion

func maxSum(nums []int) int {
	maxSum := 0
	mapSet := make(map[int]bool)

	for _, num := range nums {
		if _, exist := mapSet[num]; !exist {
			mapSet[num] = true
			if num > 0 {
				maxSum += num
			}
		}
	}
	if maxSum == 0 {
		first := true
		for key, _ := range mapSet {
			if first {
				maxSum = key
				first = false
				continue
			}
			maxSum = max(maxSum, key)
		}
		return maxSum
	}
	return maxSum
}

File: Python/Easy/3487_Maximum_Unique_Subarray_Sum_After_Deletion/solution.py
class Solution:
    def maxSum(self, nums: list[int]) -> int:
        maxSum = 0
        map = {}
        for num in nums:
            if num not in map:
                map[num] = True
                if num > 0:
                    maxSum += num
        
        if maxSum == 0:
            return max(map)
        return maxSum